exporter/awsemfexporter/metric_declaration.go

```diff
@@ -160,9 +160,6 @@ func (m *MetricDeclaration) MatchesLabels(labels map[string]string) bool {
 // returns dimensions that only contains labels from in the given label set.
 func (m *MetricDeclaration) ExtractDimensions(labels map[string]string) (dimensions [][]string) {
 	for _, dimensionSet := range m.Dimensions {
-		if len(dimensionSet) == 0 {
```

Comment:
Confused about this, you will just append an empty array to `dimensions`

Reply:
yes. empty array is a valid dimension set. 

https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html#CloudWatch_Embedded_Metric_Format_Specification_structure_dimensionset

Empty dimension set means that no dimensions will be applied to the metric. This will handle the case for metrics without label because they cannot have any dimension.